lib/internal/process/warning.js

```diff
@@ -1,78 +1,44 @@
 'use strict';
 
-const config = internalBinding('config');
+const { warningFile } = internalBinding('config');
 const prefix = `(${process.release.name}:${process.pid}) `;
 const { ERR_INVALID_ARG_TYPE } = require('internal/errors').codes;
 
 exports.setup = setupProcessWarnings;
 
-var cachedFd;
-var acquiringFd = false;
-function nop() {}
-
 // Lazily loaded
-var fs = null;
+let fs;
+let fd;
 
 function writeOut(message) {
   if (console && typeof console.error === 'function')
     return console.error(message);
   process._rawDebug(message);
 }
 
-function onClose(fd) {
-  return () => {
-    if (fs === null) fs = require('fs');
-    fs.close(fd, nop);
```

Comment:
> The file descriptor is now closed synchronously when exiting.

I’d count that as a bugfix – do you maybe want to do this particular part in a semver-patch PR?